pkg/agent/util/net_windows_test.go

```diff
@@ -0,0 +1,1342 @@
+	}
+	gotErr = SetAdapterMACAddress("test-adapter", &testMacAddr)
+	assert.Equal(t, testErr, gotErr)
+}
+
+func TestWindowsHyperVEnabled(t *testing.T) {
+	runCommand = func(cmd string) (string, error) {
+		return "Enabled_HyperV", nil
+	}
+	defer func() {
+		runCommand = ps.RunCommand
+	}()
+	gotResult, gotErr := WindowsHyperVEnabled()
+	assert.True(t, gotResult)
+	assert.NoError(t, gotErr)
+	testErr := fmt.Errorf("invalid")
+	runCommand = func(cmd string) (string, error) {
+		return "fail", testErr
+	}
+	gotResult, gotErr = WindowsHyperVEnabled()
+	assert.False(t, gotResult)
+	assert.Equal(t, testErr, gotErr)
+}
+
+func TestPrepareHNSNetwork(t *testing.T) {
+	testMacAddr, _ := net.ParseMAC("aa:bb:cc:dd:ee:ff")
+	gw, subnet, _ := net.ParseCIDR("8.8.8.8/32")
+	routes := []Route{{
+		LinkIndex:         0,
+		DestinationSubnet: subnet,
+		GatewayAddress:    gw,
+		RouteMetric:       MetricDefault,
+	}}
+	testRoutes := make([]interface{}, len(routes))
+	for i, v := range routes {
+		testRoutes[i] = v
+	}
+	tests := []struct {
+		name                   string
+		subnetCIDR             *net.IPNet
+		nodeIPNet              *net.IPNet
+		uplinkAdapter          *net.Interface
+		nodeGateway            string
+		dnsServers             string
+		routes                 []interface{}
+		newName                string
+		ipFound                bool
+		hnsNetworkCreateErr    error
+		commandErr             error
+		hnsNetworkRequestError error
+		testNetInterfaces      []*mockNetInterface
+		testNetInterfaceErr    error
+		wantErr                error
+	}{
+		{
+			name: "Prepare Success",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("8.8.8.7"),
+				Mask: net.CIDRMask(32, 32),
+			},
+			nodeIPNet: &net.IPNet{
+				IP:   ipv4Public,
+				Mask: net.CIDRMask(32, 32),
+			},
+			uplinkAdapter: &net.Interface{
+				Name:         "testAdapter",
+				Index:        0,
+				HardwareAddr: testMacAddr,
+			},
+			nodeGateway:            "testGateway",
+			dnsServers:             "192.168.1.21",
+			routes:                 testRoutes,
+			newName:                "test1",
+			ipFound:                true,
+			hnsNetworkCreateErr:    nil,
+			commandErr:             nil,
+			hnsNetworkRequestError: nil,
+			testNetInterfaceErr:    nil,
+			wantErr:                nil,
+		},
+		{
+			name: "Create Error",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("8.8.8.7"),
+				Mask: net.CIDRMask(32, 32),
+			},
+			nodeIPNet: &net.IPNet{
+				IP:   ipv4Public,
+				Mask: net.CIDRMask(32, 32),
+			},
+			uplinkAdapter: &net.Interface{
+				Name:         "testAdapter",
+				Index:        0,
+				HardwareAddr: testMacAddr,
+			},
+			nodeGateway:            "testGateway",
+			dnsServers:             "192.168.1.21",
+			routes:                 testRoutes,
+			newName:                "",
+			ipFound:                true,
+			hnsNetworkCreateErr:    fmt.Errorf("invalid"),
+			commandErr:             nil,
+			hnsNetworkRequestError: nil,
+			testNetInterfaceErr:    nil,
```

Comment:
You may just remove the field when it's default value which can make the table shorter.